src/linemod_detect.cpp

```diff
@@ -277,10 +277,13 @@ struct Detector: public object_recognition_core::db::bases::ModelReaderBase {
       }
       if (*use_depth_)
       {
+        if (depth_->depth() == CV_32F)
+          depth_->convertTo(*depth_, CV_16UC1, 1000.0);
```

Comment:
actually, there is a problem with that: you are overwriting depth_ which can be shared with different cells. Please simple create a cv::Mat depth. Code should be:

    cv::Mat depth = *depth_;
    if (depth_->depth() == CV_32F)
      depth.convertTo(depth, CV_16UC1, 1000.0);

And make sure you are using that matrix all over below. Sorry to have missed that